libs/events/src/CowEventEmitter.ts

```diff
@@ -9,12 +9,12 @@ export type CowEventListener<T extends CowEvents> = T extends keyof CowEventPayl
 export type CowEventListeners = CowEventListener<CowEvents>[]
 
 export interface CowEventEmitter {
-  on<T extends CowEvents>(event: T, handler: CowEventHandler<T>): void
+  on<T extends CowEvents>(listener: CowEventListener<T>): void
   emit<T extends CowEvents>(event: T, payload: CowEventPayloads[T]): void
   off<T extends CowEvents>(event: T, listenerToRemove: CowEventHandler<T>): void
 }
 
-export class CowEventEmitterImpl {
+export class CowEventEmitterImpl implements CowEventEmitter {
```

Comment:
Ah, thanks for already addressing my comment from the previous PR 😬 

Anyway, do we really need an interface here?
I'm thinking because I'm not happy with the Java pattern (SomethingImpl).

Got to this https://stackoverflow.com/questions/31876947/confused-about-the-interface-and-class-coding-guidelines-for-typescript

It's about the `I*` interface prefix, but I think it also applies here.

Reply:
> Anyway, do we really need an interface here?

Yes, I think so. It's nice to not depend on the specific implementation for the event emitter. This interface is using through all the project, and is what the provided exposes, and this way we don't depend on this very basic implementation i created.

> I'm thinking because I'm not happy with the Java pattern (SomethingImpl).
finee :) 
I just didn't want to artificially make a name for it, but your force me to call it ... `SimpleCowEventEmitter implements CowEventEmitter` if you find a better name, just let me know

Reply:
c085022947f439767a7d744d47040593d6dd45b5